transcode/transcode.go

```diff
@@ -120,20 +129,43 @@ func RunTranscodeProcess(transcodeRequest TranscodeSegmentRequest, streamName st
 	}
 
 	// Build the manifests and push them to storage
-	manifestManifestURL, err := GenerateAndUploadManifests(sourceManifest, targetTranscodedRenditionOutputURL.String(), transcodedStats)
+	manifestURL, err := GenerateAndUploadManifests(sourceManifest, targetTranscodedRenditionOutputURL.String(), transcodedStats)
 	if err != nil {
 		return outputs, segmentsCount, err
 	}
 
-	output := clients.OutputVideo{Type: "object_store", Manifest: manifestManifestURL}
+	playbackBaseURL := PublishDriverSession(targetTranscodedRenditionOutputURL.String(), tout.Path)
+
+	manifestURL = strings.ReplaceAll(manifestURL, targetTranscodedRenditionOutputURL.String(), playbackBaseURL)
+	output := clients.OutputVideo{Type: "object_store", Manifest: manifestURL}
 	for _, rendition := range transcodedStats {
-		output.Videos = append(output.Videos, clients.OutputVideoFile{Location: rendition.ManifestLocation, SizeBytes: int(rendition.Bytes)})
+		videoManifestURL := strings.ReplaceAll(rendition.ManifestLocation, targetTranscodedRenditionOutputURL.String(), playbackBaseURL)
+		output.Videos = append(output.Videos, clients.OutputVideoFile{Location: videoManifestURL, SizeBytes: int(rendition.Bytes)})
 	}
 	outputs = []clients.OutputVideo{output}
 	// Return outputs for .dtsh file creation
 	return outputs, segmentsCount, nil
 }
 
+// PublishDriverSession tries to publish the given osUrl and returns a publicly accessible video URL.
+// If driver supports `Publish()`, e.g. web3.storage, then return the path to the video.
+// If driver does not support `Publish()`, e.g. S3, then return the input osUrl, video should be accessible with osUrl.
+// In case of any other error, return an empty string; we don't want to propagate osUrl by default, because it may contain sensitive data.
+func PublishDriverSession(osUrl string, relPath string) string {
```

Comment:
Could we obfuscate the errors but still return an error for this function? It's weird even from the usage side for it to just manipulate some strings but is actually performing some I/O underneath that can fail. For example would we want to consider the job successful if we tried doing this publish but the call to web3.storage fails? I think we'd like to retry the job in that case as well (from here we might just need to return/propagate "an" error)

Reply:
Yes, it's a good point. Let me change it.